src/main/java/edu/berkeley/cs186/database/query/QueryPlan.java

```diff
@@ -630,22 +644,33 @@ public Map<Set<String>, QueryOperator> minCostJoins(
             Map<Set<String>, QueryOperator> prevMap,
             Map<Set<String>, QueryOperator> pass1Map) {
         Map<Set<String>, QueryOperator> result = new HashMap<>();
-        // TODO(proj3_part2): implement
-        // We provide a basic description of the logic you have to implement:
-        // For each set of tables in prevMap
-        //   For each join predicate listed in this.joinPredicates
-        //      Get the left side and the right side of the predicate (table name and column)
-        //
-        //      Case 1: The set contains left table but not right, use pass1Map
-        //              to fetch an operator to access the rightTable
-        //      Case 2: The set contains right table but not left, use pass1Map
-        //              to fetch an operator to access the leftTable.
-        //      Case 3: Otherwise, skip this join predicate and continue the loop.
-        //
-        //      Using the operator from Case 1 or 2, use minCostJoinType to
-        //      calculate the cheapest join with the new table (the one you
-        //      fetched an operator for from pass1Map) and the previously joined
-        //      tables. Then, update the result map if needed.
+        // TODO(proj3_part2): implement DONE
+
+        for (Set<String> table : prevMap.keySet()) {
```

Comment:
저도 지금 보다가 생각난건데 가독성을 위해서  이런식으로 바꿔도 좋을 것 같아요!
```suggestion
        prevMap.forEach((table, op)) -> { 
```

Reply:
람다를 쓰는 것도 가독성이 더 올라가겠네요 👍🏻